open_seq2seq/optimizers/automatic_loss_scaler.py

```diff
@@ -6,24 +6,17 @@
 
 import tensorflow as tf
 
+from open_seq2seq.utils.utils import check_params
 
 class AutomaticLossScaler(object):
   SUPPORTED_ALGOS = ['backoff', 'logmax']
 
-  def __init__(self, algorithm='Backoff', scale_min=1.0, scale_max=2.**24):
+  def __init__(self, algorithm='Backoff', params=None):
```

Comment:
If by default, params=None and user did not specify loss_scale_params (since it is optional). Then lines 64-67 will throw exception because params is None.

Reply:
if params is passed as None, then it is checked and converted to an empty.
https://github.com/blisc/OpenSeq2Seq/blob/e301a036609d0b5f376df6e014fe8bb69da78197/open_seq2seq/optimizers/automatic_loss_scaler.py#L52-L53

https://github.com/blisc/OpenSeq2Seq/blob/e301a036609d0b5f376df6e014fe8bb69da78197/open_seq2seq/optimizers/automatic_loss_scaler.py#L114-L115

I tested the changes on Backoff, but not Logmax.